core/modules/dblog/dblog.admin.inc

```diff
@@ -30,7 +30,10 @@ function dblog_overview() {
     WATCHDOG_DEPRECATED => 'dblog-deprecated',
   );
 
-  $build['dblog_clear_log_form'] = backdrop_get_form('dblog_clear_log_form');
+  if (user_access('clear log messages')) {
```

Comment:
Maybe the check would be better on the `dblog_clear_log_form` form itself (like an `#access` key on the fieldset), in case this form is called somewhere else.